static/src/javascripts/projects/commercial/modules/prebid/prebid.js

```diff
@@ -12,16 +12,84 @@ import type {
     PrebidMediaTypes,
     PrebidSlot,
 } from 'commercial/modules/prebid/types';
+import type { PrebidPriceGranularity } from 'commercial/modules/prebid/price-config';
+
+type EnableAnalyticsConfig = {
+    provider: string,
+    options: {
+        ajaxUrl: string,
+        pv: string,
+    },
+};
+
+type ConsentManagement = {
+    cmpApi: string,
+    timeout: number,
+    allowAuctionWithoutConsent: boolean,
+};
+
+type S2SConfig = {
+    accountId: string,
+    enabled: boolean,
+    bidders: Array<string>,
+    timeout: number,
+    adapter: string,
+    is_debug: string, // true or false string
+    endpoint: string,
+    syncEndpoint: string,
+    cookieSet: boolean,
+    cookiesetUrl: string,
+};
+
+type UserSync =
+    | {
+          // syncsPerBidder: number, // allow all syncs - bug https://github.com/prebid/Prebid.js/issues/2781
+          syncsPerBidder: number, // temporarily until above bug fixed
+          filterSettings: {
+              all: {
+                  bidders: string, // allow all bidders to sync by iframe or image beacons
+                  filter: string,
+              },
+          },
+      }
+    | { syncEnabled: false };
+
+type PbjsConfig = {
+    bidderTimeout: number | boolean,
+    priceGranularity: PrebidPriceGranularity,
+    userSync: UserSync,
+    consentManagement: ConsentManagement | false,
+    s2sConfig: S2SConfig,
+};
 
-const bidderTimeout = 1500;
+type XasisFunction = ({
+    appnexus: {
+        buyerMemberId: string,
+    },
+}) => string;
```

Comment:
Maybe this would be better inline or with more generic name, as doesn't apply exclusively to Xaxis.  It could be applied to any bidder.

Reply:
Yeah, I thought it was a weird name too. I'm not sure what name to give it as I don't fully understand the context it's used in, let's talk tomorrow!